### PR TITLE
documentation: Update smdp docs with sparse_as_dense support

### PR DESCRIPTION
**Pull request by ChaiBapchya**
*Issue #, if available:*
NA

*Description of changes:*
Customer pointed out the incorrect documentation related to `sparse_as_dense`
 [DistributedGradientTape](https://sagemaker.readthedocs.io/en/stable/api/training/sdp_versions/latest/smd_data_parallel_tensorflow.html?highlight=DistributedGradientTape#smdistributed.dataparallel.tensorflow.DistributedGradientTape) and [DistributedOptimizer](https://sagemaker.readthedocs.io/en/stable/api/training/sdp_versions/latest/smd_data_parallel_tensorflow.html?highlight=DistributedGradientTape#smdistributed.dataparallel.tensorflow.DistributedOptimizer)

We do support that parameter and its handled to convert tf.indexedslices to tensor [with dtype infered from value] 
https://github.com/aws/herring/blob/1719ab8a9007e6010e66a23ca201529a31a6c4e0/smdistributed/dataparallel/tensorflow/__init__.py#L289-L294

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/api/training/sdp_versions/latest/smd_data_parallel_tensorflow.rst [hunk 1]
```diff
@@ -443,7 +443,7 @@ TensorFlow API
 
       *   Supported compression types - ``none``, ``fp16``
 
-   - ``sparse_as_dense:`` Not supported. Raises not supported error.
+   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to False.
```
Comment (roywei) [inline doc/api/training/sdp_versions/latest/smd_data_parallel_tensorflow.rst:446]:
```suggestion
   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to ``False``.
```

File: doc/api/training/sdp_versions/latest/smd_data_parallel_tensorflow.rst [hunk 2]
```diff
@@ -482,6 +482,8 @@ TensorFlow API
 
       *   Supported compression types - ``none``, ``fp16``
 
+   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to False.
```
Comment (roywei) [inline doc/api/training/sdp_versions/latest/smd_data_parallel_tensorflow.rst:485]:
```suggestion
   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to ``False``.
```

File: doc/api/training/sdp_versions/v1.0.0/smd_data_parallel_tensorflow.rst [hunk 3]
```diff
@@ -456,7 +456,7 @@ TensorFlow API
 
       *   Supported compression types - ``none``, ``fp16``
 
-   - ``sparse_as_dense:`` Not supported. Raises not supported error.
+   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to False.
```
Comment (roywei) [inline doc/api/training/sdp_versions/v1.0.0/smd_data_parallel_tensorflow.rst:459]:
```suggestion
   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to ``False``.
```

File: doc/api/training/sdp_versions/v1.0.0/smd_data_parallel_tensorflow.rst [hunk 4]
```diff
@@ -496,6 +496,8 @@ TensorFlow API
 
       *   Supported compression types - ``none``, ``fp16``
 
+   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to False.
```
Comment (roywei) [inline doc/api/training/sdp_versions/v1.0.0/smd_data_parallel_tensorflow.rst:499]:
```suggestion
   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to ``False``.
```

File: doc/api/training/sdp_versions/v1.1.x/smd_data_parallel_tensorflow.rst [hunk 5]
```diff
@@ -459,7 +459,7 @@ library with TensorFlow.
 
       *   Supported compression types - ``none``, ``fp16``
 
-   - ``sparse_as_dense:`` Not supported. Raises not supported error.
+   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to False.
```
Comment (roywei) [inline doc/api/training/sdp_versions/v1.1.x/smd_data_parallel_tensorflow.rst:462]:
```suggestion
   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to ``False``.
```

File: doc/api/training/sdp_versions/v1.1.x/smd_data_parallel_tensorflow.rst [hunk 6]
```diff
@@ -499,6 +499,8 @@ library with TensorFlow.
 
       *   Supported compression types - ``none``, ``fp16``
 
+   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to False.
```
Comment (roywei) [inline doc/api/training/sdp_versions/v1.1.x/smd_data_parallel_tensorflow.rst:502]:
```suggestion
   - ``sparse_as_dense:`` Treats sparse gradient tensor as dense tensor. Defaults to ``False``.
```

**Review by roywei — DISMISSED** [review]
LGTM, minor nits to keep the same style.

**Comment by ChaiBapchya** [comment]
@ahsan-z-khan Can i get a review on this?

**Review by mchoi8739 — APPROVED** [review]

**Review by ahsan-z-khan — APPROVED** [review]

**Pull request merged**